src/main/java/org/jboss/tools/intellij/openshift/utils/odo/Component.java

```diff
@@ -10,64 +10,85 @@
  ******************************************************************************/
 package org.jboss.tools.intellij.openshift.utils.odo;
 
-import java.io.File;
-
 public interface Component {
-  String getName();
-  ComponentState getState();
-  void setState(ComponentState state);
-  String getPath();
-  void setPath(String path);
-  default boolean hasContext() {
-      return getPath() != null;
-  }
-
-  class ComponentImpl implements Component {
-    private String name;
-    private ComponentState state;
-    private String path;
-
-    private ComponentImpl(String name, ComponentState state, String path) {
-      this.name = name;
-      this.state = state;
-      this.path = path;
-    }
+    String getName();
 
-    @Override
-    public String getName() {
-      return name;
-    }
+    ComponentState getState();
 
-    @Override
-    public ComponentState getState() {
-      return state;
-    }
+    void setState(ComponentState state);
 
-    @Override
-    public void setState(ComponentState state) {
-      this.state = state;
-    }
+    String getPath();
 
-    @Override
-    public String getPath() {
-      return path;
+    void setPath(String path);
+
+    default boolean hasContext() {
+        return getPath() != null;
     }
 
-    @Override
-    public void setPath(String path) {
-      this.path = path;
+    ComponentInfo getInfo();
+
+    ComponentKind getComponentKind();
+
+    class ComponentImpl implements Component {
+        private final String name;
+        private ComponentState state;
+        private String path;
+        private final ComponentInfo info;
+        private final ComponentKind kind;
```

Comment:
Should be part of ComponentInfo

Reply:
refactored to include kind in componentInfo.